editor/src/clj/editor/resource_node.clj

```diff
@@ -42,7 +44,16 @@
    :dirty dirty
    :save-value save-value})
 
-(declare save-data-content)
+(defn save-content
+  ^String [save-value resource-type]
+  (when (some? save-value)
+    (when-let [write-fn (:write-fn resource-type)]
+      (write-fn save-value))))
+
+(defn save-data-content
+  ^String [{:keys [resource save-value]}]
+  (let [resource-type (resource/resource-type resource)]
+    (save-content save-value resource-type)))
```

Comment:
The `save-data-content` function already existed, but I broke out the `save-content` function from it and moved both up in the file, since I needed to call `save-data-content` from `produce-lines` just below this.